x-pack/platform/plugins/shared/onechat/public/application/components/agents/edit/agent_form.tsx

```diff
@@ -0,0 +1,344 @@
+import { queryKeys } from '../../../query_keys';
+import { ToolsSelection } from './tools_selection';
+
+export interface AgentFormProps {
+  mode: 'edit' | 'create';
+  agentId?: string;
+}
+
+type AgentFormData = Omit<AgentProfile, 'createdAt' | 'updatedAt'>;
+
+export const AgentForm: React.FC<AgentFormProps> = ({ mode, agentId }) => {
+  const { navigateToOnechatUrl } = useNavigation();
+  const {
+    services: { notifications },
+  } = useKibana();
+
+  const onSaveSuccess = () => {
+    notifications.toasts.addSuccess(
+      mode === 'create'
+        ? i18n.translate('xpack.onechat.agents.createSuccessMessage', {
+            defaultMessage: 'Agent created successfully',
+          })
+        : i18n.translate('xpack.onechat.agents.updateSuccessMessage', {
+            defaultMessage: 'Agent updated successfully',
+          })
+    );
+    navigateToOnechatUrl(appPaths.agents.list);
+  };
+
+  const onSaveError = (err: Error) => {
+    const errorMessage =
+      mode === 'create'
+        ? i18n.translate('xpack.onechat.agents.createErrorMessage', {
+            defaultMessage: 'Failed to create agent',
+          })
+        : i18n.translate('xpack.onechat.agents.updateErrorMessage', {
+            defaultMessage: 'Failed to update agent',
+          });
+    notifications.toasts.addError(err, {
+      title: errorMessage,
+    });
+  };
+
+  const { agentProfilesService } = useOnechatServices();
+  const queryClient = useQueryClient();
+
+  const deleteAgentMutation = useMutation({
+    mutationFn: (id: string) => {
+      if (!id) {
+        throw new Error('Agent ID is required for delete');
+      }
+      return agentProfilesService.delete(id);
+    },
+    onSuccess: (result) => {
+      queryClient.invalidateQueries({ queryKey: queryKeys.agentProfiles.byId(agentId!) });
+      queryClient.invalidateQueries({ queryKey: queryKeys.agentProfiles.all });
+      notifications.toasts.addSuccess(
+        i18n.translate('xpack.onechat.agents.deleteSuccessMessage', {
+          defaultMessage: 'Agent deleted successfully',
+        })
+      );
+      navigateToOnechatUrl(appPaths.agents.list);
+    },
+    onError: (err: Error) => {
+      notifications.toasts.addError(err, {
+        title: i18n.translate('xpack.onechat.agents.deleteErrorMessage', {
+          defaultMessage: 'Failed to delete agent',
+        }),
+      });
+    },
+  });
+
+  const {
+    state: agentState,
+    isLoading,
+    isSubmitting,
+    submit,
+    tools,
+    error,
+  } = useAgentEdit({
+    agentId,
+    onSaveSuccess,
+    onSaveError,
+  });
+
+  const formMethods = useForm<AgentFormData>({
+    defaultValues: { ...agentState },
+    mode: 'onChange',
+  });
+  const { control, handleSubmit, reset, formState, setValue, watch } = formMethods;
+
+  useEffect(() => {
+    if (agentState && !isLoading) {
+      reset(agentState);
+    }
+  }, [agentState, isLoading, reset]);
+
+  const toolSelection = watch('toolSelection');
```

Comment:
Genuine question - why do we need to do that, opposed to accessing it from `agentState` ?